source/agora/utils/TxBuilder.d

```diff
@@ -471,161 +531,240 @@ public struct TxBuilder
         return this;
     }
 
+    /// Refund output for the transaction
+    private Output leftover;
+
+    // fee per byte rate to be paid for the tx
+    private Amount fee_rate = Amount(700);
+
+    /// Stores the inputs to consume until `sign` is called
+    private OutputRef[] inputs;
+
+    /// Transactions to be built and returned
+    private Transaction data;
+
+    /// Unlocker delegate to sign the inputs
+    private Unlock delegate (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow unlocker;
+
     /***************************************************************************
 
-        Deduct all the remaining amount, so there will be no refund transaction
+        Check if leftover is large enough to cover the minimum fees
 
-        Useful when trying to test fee calculation
+        Useful when filtering to sign only valid txb with sufficient funds
 
         Returns:
-            Reference to `this` for easy chaining
+            true if funds are available to pay the minimum fees
 
     ***************************************************************************/
 
-    public ref typeof(this) deductRemaining () pure nothrow @safe @nogc
-        return
+    public bool hasMinFees () nothrow @safe
     {
-        this.leftover.value = Amount(0);
-
-        return this;
+        return this.leftover.value >= minFees(true);
     }
 
-    /// Refund output for the transaction
-    private Output leftover;
+    private Amount minFees (bool include_refund_fees = false) nothrow @safe
+    {
+        auto unlocker = this.unlocker == null ? &keyUnlocker : this.unlocker;
```

Comment:
That's a good example of a special case that can be easily avoided.

Reply:
Not so easy but have found a solution.